src/components/accessories/discharge/dischargeForm/DischargeForm.tsx

```diff
@@ -0,0 +1,291 @@
+import { useFormik } from "formik";
+import get from "lodash.get";
+import has from "lodash.has";
+import moment from "moment";
+import React, { FC, useCallback, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { object, string } from "yup";
+import warningIcon from "../../../../assets/warning-icon.png";
+import {
+  AdmissionTypeDTO,
+  DiseaseDTO,
+  DiseaseTypeDTO,
+  WardDTO,
+} from "../../../../generated";
+import {
+  differenceInDays,
+  formatAllFieldValues,
+  getFromFields,
+} from "../../../../libraries/formDataHandling/functions";
+import { getAdmissionTypes } from "../../../../state/admissionTypes/actions";
+import { getDischargeTypes } from "../../../../state/dischargeTypes/actions";
+import { getDiseasesIpdOut } from "../../../../state/diseases/actions";
+import { getWards } from "../../../../state/ward/actions";
+import { IState } from "../../../../types";
+import AutocompleteField from "../../autocompleteField/AutocompleteField";
+import Button from "../../button/Button";
+import ConfirmationDialog from "../../confirmationDialog/ConfirmationDialog";
+import DateField from "../../dateField/DateField";
+import TextField from "../../textField/TextField";
+import "./styles.scss";
+import { DischargeProps } from "./types";
+
+const DischargeForm: FC<DischargeProps> = ({
+  fields,
+  onSubmit,
+  submitButtonLabel,
+  resetButtonLabel,
+  isLoading,
+  shouldResetForm,
+  resetFormCallback,
+  admission,
+}) => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+
+  const diagnosisOutList = useSelector(
+    (state: IState) => state.diseases.diseasesIpdOut.data
+  );
+
+  const dischargeTypes = useSelector(
+    (state: IState) => state.dischargeTypes.allDischargeTypes.data
+  );
+
+  const renderOptions = (
+    data:
+      | (DiseaseDTO | AdmissionTypeDTO | DiseaseTypeDTO | DiseaseDTO)[]
+      | undefined
+  ) => {
+    if (data) {
+      return data.map((item) => {
+        return {
+          value: item.code?.toString() ?? "",
+          label: item.description ?? "",
+        };
+      });
+    } else return [];
+  };
+
+  const initialValues = getFromFields(fields, "value");
+
+  const validationSchema = object({
+    disDate: string()
+      .required(t("common.required"))
+      .test({
+        name: "disDate",
+        message: t("admission.validatelastdate", {
+          admDate: moment(admission?.admDate ?? "").format("DD/MM/YYYY"),
+        }),
+        test: function (value) {
+          return moment(value).isSameOrAfter(moment(admission?.admDate ?? ""));
+        },
+      }),
+
+    disType: string().required(t("common.required")),
```

Comment:
Both fields are not showing the validation message if I submit the form without touch them.

Reply:
This is the same for the admission form , If i cleare the form and I try to submit it not all the messages show to the user.

Reply:
Let me check it out please !